src/io/GeantImporter.hh

```diff
@@ -0,0 +1,84 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file GeantImporter.hh
+//---------------------------------------------------------------------------//
+/*!
+ * \brief Loads the geant-exporter-data.root file data into memory
+ *
+ * GeantImporter loads particle and physics table data from the ROOT file 
+ * created by the app/geant-exporter external code. Usage:
+ * 
+ * 1. The constructor takes the /path/to/rootfile.root as a parameter and 
+ *    opens the root file. 
+ * 2. Operator() can be used to load the full data or a user-defined selection
+ *    into a geant_importer_data struct that contains shared pointers to 
+ *    ParticleParams and GeantPhysicsTables.
+ *
+ * \code
+ *  GeantImporter import("/path/to/rootfile.root");
+ *  geant_importer_data full_data = import();
+ *  geant_importer_data selected_data = import({pdg::electron(), pdg::gamma()});
+ * \endcode
+ *
+ * Data from the intermediate GeantParticle class, used to write/read the ROOT
+ * file is still publicly available via a vector<GeantParticle> case needed.
+ */
+//---------------------------------------------------------------------------//
+
+#pragma once
+
+// C++
+#include <memory>
+#include <string>
+#include <vector>
+
+// Project
+#include "GeantParticle.hh"
+#include "physics/base/ParticleParams.hh"
+#include "physics/base/ParticleDef.hh"
+
+#include "base/Types.hh"
+#include "base/Macros.hh"
+
+// ROOT
+class TFile;
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+struct geant_importer_data
+{
+    std::shared_ptr<ParticleParams> particle_params;
+    // std::shared_ptr<PhysicsTables> physics_tables; // placeholder
+};
+
+class GeantImporter
+{
+  public:
+    GeantImporter(std::string filename);
+    ~GeantImporter();
+
+    geant_importer_data operator()();
+    geant_importer_data operator()(std::vector<PDGNumber> pdg_numbers);
+
+  protected:
+    void build_object_list(std::string const root_tree_name);
+    void load_particle_data();
+    // load_physics_tables(); //placeholder
+
+    std::shared_ptr<ParticleParams> get_ParticleParams();
+    // std::shared_ptr<PhysicsTable> get_PhysicsTables(); // placeholder
+
+  public:
+    std::vector<GeantParticle> particle_vector_;
+    // std::vector<GeantPhysicsTable> physics_table_vector_; // placeholder
```

Comment:
Can you remove the placeholder comments for this MR? I think there might be one or two in the exporter app as well.